tests/test_machine.py

```diff
@@ -0,0 +1,21 @@
+# SPDX-License-Identifier: LGPL-2.1+
+
+import os
+
+import mkosi.machine as machine
+import pytest
+
+
+@pytest.mark.parametrize("boot", [["-d", "fedora", "-f", "--autologin", "--bootable", "--qemu-headless", "--hostonly-initrd", "qemu"], \
+                                    ["-d", "fedora", "-f", "--autologin", "boot"], \
+                                    ["-d", "fedora", "-f", "--autologin", "shell"]])
```

Comment:
This is a bit redundant, let's just put the verb in the parametrize and then add the other arguments in the test itself based on the verb used.